k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapConnection.java

```diff
@@ -679,14 +689,19 @@ private boolean isListResponse(ImapResponse response) {
         return isListResponse && hierarchyDelimiterValid;
     }
 
-    protected boolean hasCapability(String capability) {
+    boolean hasCapability(String capability) throws IOException, MessagingException {
+        requestCapabilitiesIfNecessary();
         return capabilities.contains(capability.toUpperCase(Locale.US));
     }
 
-    public boolean isCondstoreCapable() throws IOException, MessagingException  {
+    public boolean isCondstoreCapable() throws IOException, MessagingException {
         return hasCapability(Capabilities.CONDSTORE);
     }
 
+    boolean isQresyncCapable() throws IOException, MessagingException {
```

Comment:
`isCondstoreCapable` is used in the `FolderSelectedStateCommand.getCommandLengthLimit()` method, which is in the `selectedstate` subpackage.